### PR TITLE
archive: add zstd and use all available CPU cores

### PR DESCRIPTION
**Pull request by Konfekt**

### CODE REVIEW

File: functions/archive [hunk 1]
```diff
@@ -17,12 +17,13 @@ fi
...
   (*.tar.bz|*.tar.bz2|*.tbz|*.tbz2) tar -cvjf "${@}" ;;
-  (*.tar.xz|*.txz) tar -J --help &>/dev/null && tar -cvJf "${@}" ;;
-  (*.tar.lzma|*.tlz) tar --lzma --help &>/dev/null && tar --lzma -cvf "${@}" ;;
+  (*.tar.xz|*.txz) tar -J --help &>/dev/null && env XZ_OPT=-T0 tar -cvJf "${@}" ;;
+  (*.tar.lzma|*.tlz) tar --lzma --help &>/dev/null && env XZ_OPT=-T0 tar --lzma -cvf "${@}" ;;
   (*.tar) tar -cvf "${@}" ;;
   (*.zip) zip -r "${@}" ;;
   (*.rar) rar a "${@}" ;;
   (*.7z) 7za a "${@}" ;;
+  (*.zst) zstd -c -T0 "${@}" -o "${1}" ;;
```
Comment (ericbn) [inline functions/archive]: You might want to offset the args list with `${@:2}`, so `${1}` in not included there.

Also, please include zst in the README.md.

**Review by ericbn — CHANGES_REQUESTED** [review]

File: functions/archive (hunk 1 above)
Comment (Konfekt) [reply functions/archive]: Of course. Thank you for the suggested fix.

**Comment by Konfekt** [comment]
Done.

**Comment by Konfekt** [comment]
Thank you for your work on `zimfw`, in particular for the latest improvements, turning it into a plug-in manager.

**Comment by ericbn** [comment]
Hey, and thank you for the contributions!

**Pull request merged**